sunbird-rc-esignet-integration-impl/src/test/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticaionServiceTest.java

```diff
@@ -0,0 +1,294 @@
+       try{
+            sunbirdRCAuthenticationService.initialize();
+            Assert.fail();
+        }catch (KycAuthException e){
+            Assert.assertEquals(e.getErrorCode(),"sunbird-rc authenticator field is not configured properly");
+        }
+    }
+
+    @Test
+    public void initializeWithValidIdField_thenFail() {
+        List<Map<String,String>> fieldDetailList = List.of(Map.of("id","policyNumber","type","string","format","string"));
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "idField", "policyNumber2");
+        try{
+            sunbirdRCAuthenticationService.initialize();
+            Assert.fail();
+        }catch (KycAuthException e){
+            Assert.assertEquals(e.getErrorCode(),"Invalid configuration: individual-id-field is not available in field-details.");
+        }
+    }
+
+    @Test
+    public void doKycAuthWithValidParams_thenPass() throws KycAuthException, IOException {
+        List<Map<String,String>> fieldDetailList = List.of(Map.of("id","policyNumber","type","string","format","string"),Map.of("id","fullName","type","string","format","string"));
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "idField", "policyNumber");
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "registrySearchUrl", "url");
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "entityIdField", "policyNumber");
+        // Arrange
+        String relyingPartyId = "validRelayingPartyId";
+        String clientId = "validClientId";
+        KycAuthDto kycAuthDto = new KycAuthDto(); // populate with valid data
+        AuthChallenge authChallenge=new AuthChallenge();
+        authChallenge.setFormat("string");
+        authChallenge.setAuthFactorType("KBA");
+        authChallenge.setChallenge("eyJmdWxsTmFtZSI6IlphaWQgU2lkZGlxdWUiLCJkb2IiOiIyMDAwLTA3LTI2In0=");
+
+        kycAuthDto.setChallengeList(List.of(authChallenge));
+        kycAuthDto.setIndividualId("000000");
+
+        List<Map<String,Object>> responseMap=new ArrayList<>();Map<String,Object> map=Map.of("policyNumber","654321","dob","654321");
+        responseMap.add(map);
+        ResponseEntity<List<Map<String,Object>>>  responseEntity = new ResponseEntity(responseMap, HttpStatus.OK);
+        Mockito.when(restTemplate.exchange(
+                Mockito.any(RequestEntity.class),
+                Mockito.eq(new ParameterizedTypeReference<List<Map<String,Object>>>() {})
+        )).thenReturn(responseEntity);
+
+        Map<String,String> mockChallengMap=new HashMap<>();
+        mockChallengMap.put("fullName","test");
+        Mockito.when(objectMapper.readValue(Mockito.anyString(),Mockito.eq(Map.class))).thenReturn(mockChallengMap);
```

Comment:
why should we mock the object mapper, can we not give a proper object mapper via reflection and expect it to do the covertion property ? Also eyJmdWxsTmFtZSI6IlphaWQgU2lkZGlxdWUiLCJkb2IiOiIyMDAwLTA3LTI2In0= and fullname test is not matching, so test case is not implemented correctly here

Reply:
This need to corrected for below test cases also

Reply:
reflection on objectMapper is not working